### ISSUE TITLE
`PresentationRequest.start` returns a presentation in a `connecting` state

### ISSUE THREAD

**Issue by tidoust** [w3c/web-platform-tests#3521]
@louaybassbouss, @mariuswsl, @mfoltzgoogle

While it seems to pass on Chromium, the basic Presentation API test that checks the call to `start` assumes that the `PresentationConnection` instance will be a `connected` state when the `Promise` is resolved, which I believe is incorrect:
https://github.com/w3c/web-platform-tests/blob/master/presentation-api/controlling-ua/startNewPresentation_success-manual.html#L35

If I'm reading the spec correctly, the instance will rather be in a `connecting` state:

1. The `Promise` is resolved in step 19 of the [start a presentation](http://w3c.github.io/presentation-api/#starting-a-presentation) algorithm when the presentation connection state is `connecting`. Resolving a `Promise` typically means *queueing a task* (t1) to run the `then` function.
2. The [establishment of the connection](http://w3c.github.io/presentation-api/#dfn-establish-a-presentation-connection) occurs after that and *queues another task* (t2) to change the state to `connected`.
3. t1 is queued before t2, and thus runs before as well.

In short, I think the line should be changed to:
```javascript
assert_equals(connection.state, "connecting", "The presentation is in a connecting state.");
```

Alternatively, the test could wait for the `connect` event on the `PresentationConnection` to check that the connection gets established.

**Comment by tidoust** [comment]
Note the same problem affects the defaultRequest test, since the `connectionavailable` gets fired while the connection is still connecting as well:
https://github.com/w3c/web-platform-tests/blob/master/presentation-api/controlling-ua/defaultRequest_success-manual.html#L39

**Comment by louaybassbouss** [comment]
agree with @tidoust, @mariuswsl can you fix this issue as @tidoust proposed?
